### ISSUE TITLE
Crash with rule `vue/define-emits-declaration` set to `type-literal`

### ISSUE THREAD

**Issue by Ericlm** [vuejs/eslint-plugin-vue#2335]
**Checklist**

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** eslint@8.54.0
- **eslint-plugin-vue version:** eslint-plugin-vue@9.19.1
- **Node version:** v20.8.0
- **Operating System:** macOS 14.1.1 (23B81)

**Please show your full configuration:**

```json5
/* eslint-env node */
require('@rushstack/eslint-patch/modern-module-resolution')

module.exports = {
  root: true,

  'extends': [
    'plugin:vue/vue3-essential',
    'eslint:recommended',
    '@vue/eslint-config-typescript',
    '@vue/eslint-config-prettier/skip-formatting'
  ],

  parserOptions: {
    ecmaVersion: 'latest'
  },

  rules: {
    'vue/define-emits-declaration': ['error', 'type-literal'],
  }
}
```

**What did you do?**	

```vue
<script setup lang="ts">
defineProps<{
  msg: string
}>()

// Rule is reporting bad emit 👍
// defineEmits<{
//   (e: 'change', id: number): void
//   (e: 'update', value: string): void
// }>()

// Rule is reporting bad emit 👍
// defineEmits<{
//   (e: 'update:modelValue', value: number): void
// }>()


// Rule is causing a crash 💥
// This format is preferred regarding this rule: https://typescript-eslint.io/rules/prefer-function-type/ (not installed in this project)
defineEmits<(e: 'update:modelValue', value: number) => void>()
</script>
```

**What did you expect to happen?**
The format without declaring an object should work as expected with the `'vue/define-emits-declaration': ['error', 'type-literal']` rule declared

**What actually happened?**

Running the linter actually crashes it, with the following output:
```
> vue-eslint-emits-issue@0.0.0 lint
> eslint . --ext .vue,.js,.jsx,.cjs,.mjs,.ts,.tsx,.cts,.mts --fix --ignore-path .gitignore


Oops! Something went wrong! :(

ESLint: 8.54.0

TypeError: param.members is not iterable
Occurred while linting /Users/ericlemaitre/anav/vue-eslint-emits-issue/src/components/HelloWorld.vue:20
Rule: "vue/define-emits-declaration"
    at Object.onDefineEmitsEnter (/Users/ericlemaitre/anav/vue-eslint-emits-issue/node_modules/eslint-plugin-vue/lib/rules/define-emits-declaration.js:67:44)
    at callVisitor (/Users/ericlemaitre/anav/vue-eslint-emits-issue/node_modules/eslint-plugin-vue/lib/utils/index.js:1342:21)
    at scriptSetupVisitor.CallExpression (/Users/ericlemaitre/anav/vue-eslint-emits-issue/node_modules/eslint-plugin-vue/lib/utils/index.js:1443:13)
    at ruleErrorHandler (/Users/ericlemaitre/anav/vue-eslint-emits-issue/node_modules/eslint/lib/linter/linter.js:1091:28)
    at /Users/ericlemaitre/anav/vue-eslint-emits-issue/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/Users/ericlemaitre/anav/vue-eslint-emits-issue/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/Users/ericlemaitre/anav/vue-eslint-emits-issue/node_modules/eslint/lib/linter/node-event-generator.js:297:26)
    at NodeEventGenerator.applySelectors (/Users/ericlemaitre/anav/vue-eslint-emits-issue/node_modules/eslint/lib/linter/node-event-generator.js:326:22)
    at NodeEventGenerator.enterNode (/Users/ericlemaitre/anav/vue-eslint-emits-issue/node_modules/eslint/lib/linter/node-event-generator.js:340:14)
```

**Repository to reproduce this issue**

The repository is here: https://github.com/Ericlm/eslint-vue-emits-issue/tree/main
You just need to clone it, install deps, and then go to `src/components/HelloWorld.vue`. The script of this file contains three different emits, that should all be reported by the linter. The first two work fine, but the last one is causing the issue.

Thank you! :)

**Comment by FloEdelmann** [comment]
CC @Meesayen

**Comment by Meesayen** [comment]
@ota-meshi  thanks for the quick fix. I learned today of the existence of that function type syntax for defineEmits :/ my bad

**Comment by FloEdelmann** [comment]
No worries, mistakes happen :slightly_smiling_face:

**Comment by Ericlm** [comment]
Thank you for the quick fix ! 🏎️👍
